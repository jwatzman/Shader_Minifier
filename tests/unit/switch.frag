void switchConst() {
  switch (42) {
    case 42:
      break;
  }
}

#define GOOD 42
void switchDefine() {
  switch (42) {
    case GOOD:
      break;
  }
}

void switchArg(in int someArg) {
  switch (someArg) {
    case 0:
      break;
  }
}

void switchExpr(in int someArg) {
  switch (someArg + 42) {
    case 0:
      break;
  }
}

float switchMultiple() {
  switch (42) {
    case 42:
      float someVar = 4.2;
      float otherVar = 2.4;
      return someVar + otherVar;
    case 43:
      float someVar = 4.3;
      return someVar + 3.4;
    default:
      return 0.0;
  }
}
