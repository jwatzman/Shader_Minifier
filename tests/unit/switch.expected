void h()
{
  switch(42){
    case 42:
      break;
  }
}

#define GOOD 42

void i()
{
  switch(42){
    case GOOD:
      break;
  }
}
void h(in int i)
{
  switch(i){
    case 0:
      break;
  }
}
void i(in int i)
{
  switch(i+42){
    case 0:
      break;
  }
}
float G()
{
  switch(42){
    case 42:
      float i=4.2;
      float k=2.4;
      return i+k;
    case 43:
      float G=4.3;
      return G+3.4;
    default:
      return 0.;
  }
}
